prac_05/emails.py

```diff
@@ -0,0 +1,30 @@
+"""
+User Email Storage
+Estimate: 25 minutes
+Actual:   35 minutes
+"""
+
+def extract_name(email):
+    """Extracts name from email address."""
+    name_parts = email.split('@')[0].split('.')
+    name = ' '.join(name_parts).title()
+    return name
+
+# Initialize dictionary to store emails and names
+user_info = {}
+
+# Input loop
+email = input("Email: ")
+while email != "":
+    name_from_email = extract_name(email)
+    user_name = input(f"Is your name {name_from_email}? (Y/n) ").strip().lower()
```

Comment:
I believe this input is not the "user_name" but rather a confirmation from user. Either Y (or) n (or) user clicks enter and then this variable stores it.